LLama/LLamaWeights.cs

```diff
@@ -165,5 +166,35 @@ public LLamaToken[] Tokenize(string text, bool add_bos, bool special, Encoding e
         {
             return NativeHandle.Tokenize(text, add_bos, special, encoding);
         }
+
+        /// <summary>
+        /// Count the tokens in the input text
+        /// </summary>
+        /// <param name="text">input text</param>
+        /// <param name="parameters">context parameters</param>
+        /// <returns></returns>
+        public int CountTokens(string text, IContextParams parameters)
+        {
+            using var context = CreateContext(parameters);
+            var count = context.Tokenize(text, special: true).Length;
+            return count;
```

Comment:
There's a `Tokenize` method on the lower level model handle, no need to use a context: https://github.com/SciSharp/LLamaSharp/blob/master/LLama/Native/SafeLlamaModelHandle.cs#L480

Reply:
(i.e. just writing a wrapper over that method should suffice)

Reply:
I did not realize that there is a tokenize method on parent level. I will use that then and since it does not need a context I can move all code back to KM. The main aim of this PR was to remove the context that was create on several places because it unnecessarily fills GPU memory (saves about 30% of memory!).